### PR TITLE
Stops fixes

### PR DESCRIPTION
**Pull request by kahummer**

### CODE REVIEW

File: src/connectors/ona-api/aggregateFormData.js [hunk 1]
```diff
@@ -9,9 +9,17 @@ function processFormData(formData, indicatorField, aggregateOptions) {
...
-  const possibleDateFormats = ['YYYY-MM-DD', 'MM/DD/YYYY'];
+  const possibleDateFormats = [
+    'YYYY-MM-DD',
+    'MM/DD/YYYY',
+    'MM-DD-YYYY',
+    'YYYY/MM/DD',
+    'YYYY/DD/MM',
+    moment.ISO_8601,
+    moment.CUSTOM_FORMAT
+  ];
```
Comment (cKellyDesign) [inline src/connectors/ona-api/aggregateFormData.js]: Nice, good call!

File: src/connectors/ona-api/aggregateFormData.js [hunk 2]
```diff
@@ -9,9 +9,17 @@ function processFormData(formData, indicatorField, aggregateOptions) {
...
 
-  const isCumulative = aggregateOptions.timeseries.type === 'cumulative';
+  const isCumulative = aggregateOptions.timeseries && aggregateOptions.timeseries.type === 'cumulative';
```
Comment (cKellyDesign) [inline src/connectors/ona-api/aggregateFormData.js]: Good catch

File: src/connectors/ona-api/aggregateFormData.js [hunk 3]
```diff
@@ -22,15 +30,15 @@ function processFormData(formData, indicatorField, aggregateOptions) {
 
   // Add week number to data
   if (isUsingToday) {
-    data = data.map((datum) => {
+    data = (data.features || data).map((datum) => {
```
Comment (cKellyDesign) [inline src/connectors/ona-api/aggregateFormData.js]: 👍

File: src/connectors/ona-api/aggregateFormData.js [hunk 4]
```diff
@@ -91,6 +107,7 @@ function processFormData(formData, indicatorField, aggregateOptions) {
   }
 
   if (isUsingToday) {
+    console.log("avai periods", availablePeriods);
```
Comment (cKellyDesign) [inline src/connectors/ona-api/aggregateFormData.js]: Let's remove console logs

File: src/map/addLayer.js [hunk 5]
```diff
@@ -38,7 +38,7 @@ function buildRadiusAsDistanceExpression(layer) {
 export default function (layer, mapConfig) {
   const layerObj = { ...layer };
   layerObj.filters = layerObj.filters || {};
-  const timefield = (layer.aggregate && layer.aggregate.timeseries) ? layer.aggregate.timeseries.field : '';
+  const timefield = (layer.aggregate && layer.aggregate.timeseries) ? layer.aggregate.timeseries.field : null;
```
Comment (cKellyDesign) [inline src/map/addLayer.js]: Will this ever render as `null` in the slider component?

File: src/map/addLayer.js [hunk 6]
```diff
@@ -119,7 +119,11 @@ export default function (layer, mapConfig) {
 
     // override from layers.json
     if (layer.paint) {
-      styleSpec.paint = layer.paint;
+      styleSpec.paint = Object.assign(
+        {},
+        styleSpec.paint,
+        layer.paint
+      );
```
Comment (cKellyDesign) [inline src/map/addLayer.js]: Nice, no mutations!

File: src/map/addLayer.js [hunk 7]
```diff
@@ -52,7 +52,7 @@ export default function (layer, mapConfig) {
     layerObj.legendBottom = 40;
   }
 
-  if (layer.property) {
+  if (layer.property && layer.categories) {
```
Comment (cKellyDesign) [inline src/map/addLayer.js]: Interesting, has `layer.categories` been a requirement for stops this whole time??

File: src/map/buildTimeseriesData.js [hunk 8]
```diff
@@ -38,7 +38,8 @@ export default function buildTimeseriesData(
   const periodDataFilter = (p) => {
     // define actual period data
     periodData[p] = {
-      data: layerObj.source.data.filter(d => d[layerObj.aggregate.timeseries.field] === p),
+      data: (layerObj.source.data.features||layerObj.source.data).filter(d =>
+        (d.properties||d)[layerObj.aggregate.timeseries.field] === p),
```
Comment (cKellyDesign) [inline src/map/buildTimeseriesData.js]: 👏

File: src/map/generateStops.js [hunk 9]
```diff
@@ -107,7 +110,9 @@ function getStops(layer) {
         if (periodProp[m] === uniqPeriods[n]) {
           periodStops[n].push(colorsStops[m]);
           periodRadius[n].push(radiusStops[m]);
-          periodStroke[n].push([radiusStops[m][0], 1]);
+          if (typeof radiusStops[m] !== 'undefined' && radiusStops[m] > 0) {
+            periodStroke[n].push([radiusStops[m][0], 1]);
+          }
```
Comment (cKellyDesign) [inline src/map/generateStops.js]: Oh snap, has this been the spot that's been plaguing us for so long??

File: src/map/prepareLayer.js [hunk 10]
```diff
@@ -276,9 +276,23 @@ function fetchMultipleSources(mapId, layer, dispatch) {
...
+      const mData = [...(mergedData.features || mergedData)];
+      if (Array.isArray(mergedData)) {
+        mergedData = mData.filter(d => d.hasOwnProperty(layerObj.property));
+      } else {
+        mergedData = {
+          ...mergedData,
+          features: mData.filter(d => d.properties.hasOwnProperty(layerObj.property)),
+        };
+      }
+    }
```
Comment (cKellyDesign) [inline src/map/prepareLayer.js]: Oh very interesting, nice

File: src/utils/groupBy.js [hunk 11]
```diff
@@ -1,7 +1,7 @@
 export default function (xs, key) {
   return xs.reduce((r, x) => {
     const rv = r;
-    (rv[x[key]] = rv[x[key]] || []).push(x);
+      (rv[x[key]] = rv[x[key]] || []).push(x);
```
Comment (cKellyDesign) [inline src/utils/groupBy.js]: Let's keep the tabbing how it was 🙏

**Comment by cKellyDesign** [comment]
@kahummer does this PR have an Issue?

**Comment by kahummer** [comment]
@cKellyDesign  I don't think it has an issue. Seems to be addressing [data-parse, isUsingToday] layer spec props and stops generation for live layers

**Pull request closed without merging**